### PR TITLE
fix(docs): add scheduler to custom provider example

### LINKED ISSUE #25463: 🐛 Bug Report: Custom Entity Providers documentation incomplete
### 📜 Description

Referencing my issue #25404, from my work account, the [Custom Entity Providers](https://backstage.io/docs/features/software-catalog/external-integrations/#custom-entity-providers) doc is incomplete and needs to specify properly what you should do in the new Backend.

After following the guide and creating the `FrobsProvider` class in the old Backend you [install the provider](https://backstage.io/docs/features/software-catalog/external-integrations/#installing-the-provider) like so 
```
  const frobs = new FrobsProvider('production', env.reader);
  builder.addEntityProvider(frobs);

  const { processingEngine, router } = await builder.build();
  await processingEngine.start();

  await env.scheduler.scheduleTask({
    id: 'run_frobs_refresh',
    fn: async () => {
      await frobs.run();
    },
    frequency: { minutes: 30 },
    timeout: { minutes: 10 },
  });
```

while in the [new Backend](https://backstage.io/docs/features/software-catalog/external-integrations/#new-backend-system) it say to do it like so:
```
export const catalogModuleFrobsProvider = createBackendModule({
  pluginId: 'catalog',
  moduleId: 'frobs-provider',
  register(env) {
    env.registerInit({
      deps: {
        catalog: catalogProcessingExtensionPoint,
        reader: coreServices.urlReader,
      },
      async init({ catalog, reader }) {
        catalog.addEntityProvider(new FrobsProvider('dev', reader));
      },
    });
  },
});
```

As you can see, the whole scheduler is missing but especially this bit of code inside the scheduler is missing `await frobs.run();` in the new Backend and there is no mention of how to add it

### 👍 Expected behavior

The new Backend documentation should mention how to trigger the `run` function after installation, and to be completed, it would be nice to add also how you can add a scheduler.

### 👎 Actual Behavior with Screenshots

After following the documentation you end up with the new entity provider not working and without knowing the real reason why only the logger inside this function is triggered.

```
  async connect(connection: EntityProviderConnection): Promise<void> {
    this.logger.info('Establishing connection.');
    this.connection = connection;
  }
```

### 👟 Reproduction steps

- migrate the catalog to the new Backend
- add a new entity provider following the documentation above

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by aramissennyeydd**
## Hey, I just made a Pull Request!

Fixes #25463

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — COMMENTED** [review]
Yes but ... :)

Currently the scheduler may start its first task invocation basically immediately, even before `connect` has been called, which is unfortunate (even in an example).

1. The hacky fix is to add an `initialDelay` and hope that it suffices.
2. The better way is to instead use `scheduler.createScheduledTaskRunner` and pass that `TaskRunner` to the `FrobsProvider`, which then can call its `run` inside `connect`.
3. The even better way is to show how to store the schedule in config and read from that.

I'd say going for 2 in an example is plenty enough.

**Comment by sebalaini** [comment]
hey @freben if you take a look at my entity provider in issue #25404 it looks like the only thing called once the provider is installed is only the `connect`, or at least from all the logger I have that one was the only one appearing in the terminal.

**Comment by aramissennyeydd** [comment]
@camilaibs I can take a look, does that same approach work for the old backend? This section of the docs still has both.

**Comment by freben** [comment]
Yeah should be the same calls. The new backend system at the end of the day mainly deals with dependency injection; the actual services and stuff is mostly exactly the same

File: docs/features/software-catalog/external-integrations.md [hunk 1]
```diff
@@ -300,9 +306,17 @@ export const catalogModuleFrobsProvider = createBackendModule({
       deps: {
         catalog: catalogProcessingExtensionPoint,
         reader: coreServices.urlReader,
+        /* highlight-add-start */
+        scheduler: coreServices.scheduler,
+        /* highlight-add-end */
       },
-      async init({ catalog, reader }) {
-        catalog.addEntityProvider(new FrobsProvider('dev', reader));
+      async init({ catalog, reader, scheduler, config }) {
```
Comment (aramissennyeydd) [inline docs/features/software-catalog/external-integrations.md:313]:
```suggestion
      async init({ catalog, reader, scheduler }) {
```

**Comment by aramissennyeydd** [comment]
@freben This should be updated with the latter 2 of your recommendations!

**Review by freben — APPROVED** [review]
Sweet!

**Comment by sebalaini** [comment]
TY so much for this ❤️

File: docs/features/software-catalog/external-integrations.md [hunk 2]
```diff
@@ -318,6 +332,98 @@ backend.add(catalogModuleFrobsProvider);
 backend.start();
 ```
 
+#### Follow-up: Config Defined Schedule
```
Comment (awanlin) [inline docs/features/software-catalog/external-integrations.md:335]: Should we also mention using the `config.d.ts` file here? Could be as simple as pointing to the docs on the topic

**Pull request merged**